include/rdma/fi_ext.h

```diff
@@ -124,6 +125,32 @@ struct fid_mem_monitor {
 };
 
 
+/* Peer provider CQ support. */
+struct fid_peer_cq;
+
+struct fi_ops_cq_write {
+	size_t	size;
+	ssize_t (*write)(struct fid_peer_cq *cq, void *context, uint64_t flags,
+			size_t len, void *buf, uint64_t data, uint64_t tag,
+			fi_addr_t src);
+	ssize_t	(*writeerr)(struct fid_peer_cq *cq,
+			const struct fi_cq_err_entry *err_entry);
+};
+
+struct fi_ops_cq_progress {
+	size_t	size;
+	void	(*progress)(struct fid_peer_cq *cq);
```

Comment:
Maybe it is better for progress to return number of CQ entry it has written? Sometime we want to keep the progress engine running until something has been completed.

Reply:
I see. Yes, it make sense. The CQ exporter can keep a counter in its CQ.